### PR TITLE
DI-240 Create EventBridge to send API change request to DoS

### PR DESCRIPTION
**Pull request by JackPlowman**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-240

## Description

This PR is to replace the event sender with AWS Eventbridge API Destination.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notable Change

- New API Gateway called change-request-receiver
- Remove useless utilities functions

## Testing

Please tick the testing that has been completed

- [ ] Unit tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by mabe13 — APPROVED** [review]
Looks good, another than it seem that the correlation id is no longer being passed through the api endpoint. Based on what this ticket changes it doesn't seem like it is the cause. So once it's passed testing it should be ok to merge 👍 

There a few other minor few things I want to discuss with Jack when he's back but they should stop the ticket being merged

**Review by mabe13 — APPROVED** [review]

**Review by ed-sparkes — CHANGES_REQUESTED** [review]

**Comment by mabe13** [comment]
When I ran the unit tests I found 4 tests were failing, are the unit test on this branch failing for anyone else ?

**Comment by ed-sparkes** [comment]
> When I ran the unit tests I found 4 tests were failing, are the unit test on this branch failing for anyone else ?

Yes i get 8 failing !

File: application/common/tests/test_utilities.py [hunk 1]
```diff
@@ -4,66 +4,28 @@
 from aws_lambda_powertools.utilities.data_classes.sqs_event import SQSRecord
 
 from pytest import raises
-
+from application.common import utilities
```
Comment (JackPlowman) [inline application/common/tests/test_utilities.py:7]: Why does need to be imported? Surely `from ..utilities import` does the same

File: application/common/tests/test_utilities.py [hunk 2]
```diff
@@ -90,7 +52,7 @@ def test_is_mock_mode_none():
     assert response == expected_response
 
 
-@patch(f"{FILE_PATH}.client")
+@patch.object(utilities, "client")
```
Comment (JackPlowman) [inline application/common/tests/test_utilities.py:55]: This function can be deleted

File: application/common/utilities.py [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 from aws_lambda_powertools import Logger
 from aws_lambda_powertools.utilities.data_classes.sqs_event import SQSRecord
 from boto3 import client
 
-logger = Logger(child=True)
+
+logger = Logger()
 
 
 def is_debug_mode() -> bool:
```
Comment (JackPlowman) [inline application/common/utilities.py:12]: This function can be deleted as it isn't used elsewhere

Comment (JackPlowman) [reply application/common/utilities.py:12]: Same with most functions in this file `is_mock_mode, get_environment_variable, invoke_lambda_function`

File: infrastructure/stacks/lambda-iam-roles/main.tf [hunk 4]
```diff
@@ -52,7 +52,9 @@ resource "aws_iam_role_policy" "event_processor_policy" {
         "ec2:DescribeVpcs",
         "xray:PutTraceSegments",
         "xray:PutTelemetryRecords",
-        "lambda:InvokeFunction"
+        "lambda:InvokeFunction",
```
Comment (JackPlowman) [inline infrastructure/stacks/lambda-iam-roles/main.tf:55]: "lambda:InvokeFunction" permission isn't required

**Review by ed-sparkes — DISMISSED** [review]

File: infrastructure/stacks/lambda-iam-roles/main.tf (hunk 4 above)
Comment (JackPlowman) [reply infrastructure/stacks/lambda-iam-roles/main.tf:55]: Also SQS permissions aren't required as they are in an allow block below

**Review by JackPlowman — COMMENTED** [review]
I am requesting these change but can't request changes on a PR I created

File: application/common/tests/test_utilities.py (hunk 1 above)
Comment (mabe13) [reply application/common/tests/test_utilities.py:7]: That's needed for the patch on line 55 which use the utiliites module as an object patch

**Comment by JackPlowman** [comment]
LGTM 👍

**Review by ed-sparkes — APPROVED** [review]

**Review by mabe13 — APPROVED** [review]

**Pull request merged**